client/src/App.js

```diff
@@ -7,15 +7,58 @@ import Footer from "./components/Footer";
 import Confirm from "./components/Confirm";
 import Calendar from "./components/Calendar";
 import Reports from "./components/Reports";
+import Add from "./components/Add";
+import { addHelper } from "./utils/addHelper.js";
+
 import { BrowserRouter, Route } from "react-router-dom";
 import data from "./data.json";
 import "typeface-roboto";
 
 class App extends Component {
   state = {
-    carData: null
+    carData: 1,
+    purpose: "Personal",
+    driver_name: null,
+    start_km: null,
+    end_km: null,
+    total: null,
+    note: null,
+    failed:false
   };
 
+  getlastkm =()=>{
```

Comment:
Is this needed? I thought this was going to be stored in the car table anyway, so won't we already have this information from when we generate the car list?

Reply:
- because we will save a `car_id`  in next step , and we do not know if another user change it to same car .